diff --git a/lib/src/navigation_rail.dart b/lib/src/navigation_rail.dart
--- a/lib/src/navigation_rail.dart
+++ b/lib/src/navigation_rail.dart
@@ -955,7 +955,9 @@
     final Color effectiveSplashColor = primaryColorAlphaModified
         ? colors.primary
         : colors.primary.withOpacity(0.12);
-    final Color effectiveHoverColor = primaryColorAlphaModified
+    final Color effectiveHoverColor = Theme.of(context).hoverColor.a == 0
+        ? Theme.of(context).hoverColor
+        : primaryColorAlphaModified
         ? colors.primary
         : colors.primary.withOpacity(0.04);
     return Semantics(
